posts/2024-05-05-liberty-project-starter-guide-IntelliJ.adoc

```diff
@@ -0,0 +1,209 @@
+
+[role='command']
+```
+export JAVA_HOME=<your Java installation path>
+```
+
+=== Installing Liberty Tools for IDE Enhancement
+Liberty Tools is necessary to install before proceeding with this blog. Please use the link:https://plugins.jetbrains.com/plugin/14856-liberty-tools[link] to install Liberty Tools on IntelliJ IDEA.
+
+[#libertyStarter]
+== Creating a Liberty Maven project with the Open Liberty Starter
+
+The Open Liberty Starter provides a simple way for initiating an Liberty Maven project. Visit the link:https://openliberty.io/start/[Open Liberty Starter page], define your application and project name, select a build tool (either Maven), and indicate the preferred versions of Java SE, Jakarta EE, and MicroProfile for your application. Finally, click on `Generate Project` to download your project seamlessly.
+
+image::img/blog/liberty-starter.png[Maven dependencies example,width=60%,align="center"]
+
+[#ImportProject]
+== Importing the project into the IntelliJ IDEA
+
+When using IntelliJ IDEA, you can either navigate to `File` > `Open...` within the IDE or simply drag and drop the project folder into the IDE window for seamless integration.
+
+[#AboutProject]
+== Overview of the Liberty Maven project
+
+=== Project structure
+
+A well-organized file structure is crucial for Maven projects, providing a clear framework for development. This hierarchy includes directories for application code, MicroProfile, Liberty configuration and tests.
+
+image::img/blog/liberty-app-directory-img.png[Liberty Project directory image,width=40%,align="center"]
+
+Referring to the demo Maven project's structure shown in the picture: 
+
+- `src/main/java`: Java application code files
+- `src/main/liberty/config`: Liberty configuration files
+- `src/main/resources/META-INF`: MicroProfile configuration files
+- `src/test`: Test files
+- `Dockerfile`: Dockerfile for building the Docker image
+- `mvnw`/`mvnw.cmd`: Maven Wrapper script for Unix-like/Windows systems
+
+
+Within the `app-name` directory, the `pom.xml` file containing configuration details for the project, encompassing dependencies, plugins, and other settings.
+
+=== Declaring dependencies
+Maven manage the project through the Maven configuration `pom.xml` file. To declare dependencies, `<dependencies>` section should be used. For example:
+
+[source]
+----
+<dependencies>
+    <dependency>
+        <groupId>jakarta.platform</groupId>
+        <artifactId>jakarta.jakartaee-api</artifactId>
+        <version>10.0.0</version>
+        <scope>provided</scope>
+    </dependency>
+</dependencies>
+----
+
+In this example, the `jakarta.jakartaee-api` from `jakarta.platform` has been introduced as a dependency for this project. 
```

Comment:
Thank you for the suggestion, according to [Naming Conventions](https://maven.apache.org/guides/mini/guide-naming-conventions.html) on maven.apache.org, the `groupId` `jakarta.platform` is used to identify the Jakarta EE project across all projects. The `package` usually refers to the root Java package containing all generated source codes, I think it might makes more sense to consider this `groupId` as an identifier for the project. Perhaps we may consider using `jakarta.platform` project? Please let me know what you think :)

Reply:
Yes, that makes sense.